### PR TITLE
replace webpack with esbuild and update phoenix to 1.6

### PR DESCRIPTION
**Pull request by WanjikuMac**
To test on local go to http://localhost:8080/

### CODE REVIEW

File: lib/twinkly_maha_web/endpoint.ex [hunk 1]
```diff
@@ -24,7 +24,8 @@ defmodule TwinklyMahaWeb.Endpoint do
     at: "/",
     from: :twinkly_maha,
     gzip: false,
-    only: ~w(css fonts images js favicon.ico robots.txt)
+    only: ~w(assets fonts images favicon.ico robots.txt)
+    # only: ~w(css fonts images js favicon.ico robots.txt)
```
Comment (sigu) [inline lib/twinkly_maha_web/endpoint.ex:28]: Remove the commented line

File: mix.exs [hunk 2]
```diff
@@ -41,24 +41,25 @@ defmodule TwinklyMaha.MixProject do
   defp deps do
     [
       {:credo, "~> 1.4", only: [:dev, :test], runtime: false},
-      {:ecto, ">= 3.4.4 and < 3.4.7"},
+      {:ecto, ">= 3.4.4 and < 3.4.7", override: true},
```
Comment (sigu) [inline mix.exs:44]: Why do we need to override ecto?

**Review by sigu — CHANGES_REQUESTED** [review]
Are we still using nodejs?
Do we need to have the `npgroress` package?

File: mix.exs (hunk 2 above)
Comment (WanjikuMac) [reply mix.exs:44]: because it was throwing an error compilation time, complaining it's not compatible with I think phoenix 1.5

Comment (sigu) [reply mix.exs:44]: What if we upgrde ecto too? 🤔 , we should not be having a lot of broken implementation on the codebase

Comment (WanjikuMac) [reply mix.exs:44]: Okay, I can try this

File: config/config.exs [hunk 3]
```diff
@@ -26,6 +26,14 @@ config :logger, :console,
 # Use Jason for JSON parsing in Phoenix
 config :phoenix, :json_library, Jason
 
+#esbuild config 
+config :esbuild,
+  version: "0.12.18",
```
Comment (TraceyOnim) [inline config/config.exs:31]: Use v0.14 and above for esbuild, v0.12.18 is too behind https://www.npmjs.com/package/esbuild

**Comment by TraceyOnim** [comment]
Don't forget to replace `RUN mix phx.digest` with ` RUN mix assets.deploy` in dockerfile

**Comment by TraceyOnim** [comment]
I think this files should be removed 
` assets/package.json, assets/package-lock.json `
Did you remove `node_modules` ? I think you might have forgotten to push the changes since its `gitignored`

**Comment by TraceyOnim** [comment]
Images in `assets/static` should be transferred to `priv/static`

**Comment by WanjikuMac** [comment]
> Don't forget to replace `RUN mix phx.digest` with ` RUN mix assets.deploy` in dockerfile

@TraceyOnim This now makes sense, when I read this https://hexdocs.pm/phoenix/deployment.html#compiling-your-application-assets

**Comment by TraceyOnim** [comment]
I can only see deleted images. Did you move the images to `priv/static`? 
If you did, you also need to push the changes . In `.gitignore file`  , `/priv/static` has been ignored . Replace that line with `/priv/static/assets` so that you can only ignore assets and exclude the images

**Comment by WanjikuMac** [comment]
> I can only see deleted images. Did you move the images to `priv/static`?
> If you did, you also need to push the changes . In `.gitignore file` , `/priv/static` has been ignored . Replace that line with `/priv/static/assets` so that you can only ignore assets and exclude the images

@TraceyOnim This should be now good to go

**Comment by WanjikuMac** [comment]
> Are we still using nodejs? Do we need to have the `npgroress` package?

This has been updated

**Pull request merged**